### PR TITLE
Update Dockerfile to do multistage builds

### PR DESCRIPTION
**Pull request by ben-toogood**

### CODE REVIEW

**Comment by ben-toogood** [comment]
> I would make the entrypoint dumb-init, so the args[] is the service to run

Done @jakexks

**Comment by asim** [comment]
Can't remember but I think scratch image will have issues with ca certs, etc. @jakexks would alpine be better here?

File: Dockerfile [hunk 1]
```diff
@@ -1,14 +1,25 @@
...
+# Copy the services
+FROM micro/go-micro as builder
+COPY . /services
+WORKDIR services
+RUN mkdir builds
+
+# Build the services
+RUN for dir in */; do \
+cd $dir && \
+(CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build . && mv ./${dir::-1} ../builds/${dir::-1} || true) && \
```
Comment (asim) [inline Dockerfile]: Would it just be better to run `make build` in every directly and keep it clean?

**Comment by jakexks** [comment]
If you just want CA certs Google's distroless/static container is better than scratch https://github.com/GoogleContainerTools/distroless/blob/master/base/README.md

**Comment by asim** [comment]
Let's pick something that's going to work across all edge cases so we don't have to constantly patch this.

File: Dockerfile (hunk 1 above)
Comment (ben-toogood) [reply Dockerfile]: make build doesn't have the correct options (goarch etc). I could pass them into the make build command, but I think this is more explicit.

**Pull request merged**